app/resources/cdl/charge_manager.rb

```diff
@@ -10,6 +10,22 @@ def initialize(resource_id:, eligible_item_service:, change_set_persister:)
       @resource_id = resource_id
       @eligible_item_service = eligible_item_service
       @change_set_persister = change_set_persister
+      clean_expired_charges
+    end
+
+    def clean_expired_charges
+      resource_charge_list.charged_items = resource_charge_list.charged_items.map do |item|
+        if expired?(item)
+          change_set_persister.delete(change_set: ChargedItemChangeSet.new(item))
```

Comment:
ChargedItems aren't persisted, they're just embedded, so I don't think you need to do this.